server/v2/stf/branch/bench_test.go

```diff
@@ -0,0 +1,172 @@
+package branch
+
+import (
+	"testing"
+
+	"cosmossdk.io/core/store"
+	coretesting "cosmossdk.io/core/testing"
+)
+
+func Benchmark_CacheStack1_Set(b *testing.B) {
+	bs := makeBranchStack(b, 1)
+	b.ResetTimer()
+	b.ReportAllocs()
+	// test sets
+	for i := 0; i < b.N; i++ {
+		err := bs.Set([]byte{0}, []byte{0})
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+func Benchmark_CacheStack10_Set(b *testing.B) {
+	bs := makeBranchStack(b, 10)
+	b.ResetTimer()
+	b.ReportAllocs()
+	// test sets
+	for i := 0; i < b.N; i++ {
+		err := bs.Set([]byte{0}, []byte{0})
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+func Benchmark_CacheStack100_Set(b *testing.B) {
+	bs := makeBranchStack(b, 100)
+	b.ResetTimer()
+	b.ReportAllocs()
+	// test sets
+	for i := 0; i < b.N; i++ {
+		err := bs.Set([]byte{0}, []byte{0})
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
```

Comment:
we should be able to combine all these into 1. potentially an array of values, also can we do 100, 1000, 10000 instead. I think it would display better how things would work on a chain